Orange/widgets/data/owselectrows.py

```diff
@@ -809,5 +880,73 @@ def resizeEvent(self, QResizeEvent):
         self.set_text()
 
 
+class DateTimeWidget(QDateTimeEdit):
+    def __init__(self, parent, col_idx, datetime_format):
+        QDateTimeEdit.__init__(self, parent)
+
+        self.parent = parent
+        self.format = datetime_format
+        self.have_date, self.have_time = datetime_format[0], datetime_format[1]
+        self.column = parent.data[:, col_idx]
+        self.set_format()
+        self.setSizePolicy(
+            QSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding))
+        self.dateTimeChanged.connect(parent._invalidate_dates)
+
+    def set_format(self):
+        str_format = Qt.ISODate
+        if self.have_date and self.have_time:
+            self.setDisplayFormat("yyyy-MM-dd hh:mm:ss")
+            c_format = "%Y-%m-%d %H:%M:%S"
+            min_datetime, max_datetime = self.find_range(self.column, c_format)
+            self.min_datetime = QDateTime.fromString(min_datetime, str_format)
+            self.max_datetime = QDateTime.fromString(max_datetime, str_format)
+            self.setCalendarPopup(True)
+            self.calendarWidget = CalendarWidgetWithTime(self)
+            self.setCalendarWidget(self._calendarWidget)
+            self.setDateTimeRange(self.min_datetime, self.max_datetime)
+
+        elif self.have_date and not self.have_time:
+            self.setDisplayFormat("yyyy-MM-dd")
+            self.setCalendarPopup(True)
+            min_datetime, max_datetime = self.find_range(self.column, "%Y-%m-%d")
+            self.min_datetime = QDate.fromString(min_datetime, str_format)
+            self.max_datetime = QDate.fromString(max_datetime, str_format)
+            self.setDateRange(self.min_datetime, self.max_datetime)
+
+        elif not self.have_date and self.have_time:
+            self.setDisplayFormat("hh:mm:ss")
+            min_datetime, max_datetime = self.find_range(self.column, "%H:%M:%S")
+            self.min_datetime = QTime.fromString(min_datetime, str_format)
+            self.max_datetime = QTime.fromString(max_datetime, str_format)
+            self.setTimeRange(self.min_datetime, self.max_datetime)
+
+    def set_datetime(self, datetime, time=None):
+        if self.have_date and self.have_time:
+            if datetime and time:
+                self.setDateTime(QDateTime(datetime, time))
+            else:
+                self.setDateTime(datetime if datetime else self.min_datetime)
+        elif self.have_date and not self.have_time:
+            self.setDate(datetime if datetime else self.min_datetime)
+        elif not self.have_date and self.have_time:
+            self.setTime(datetime if datetime else self.min_datetime)
+        self.dateTimeChanged.connect(self.parent.conditions_changed)
```

Comment:
Did you mean to connect the signal or emit it? I suppose you wanted to do this:

```python
self.dateTimeChanged.emit()
```

To call (or connect) the proper method is up to the parent widget.